File: leetcode_38.cpp
class Solution {
public:
  string longestCommonPrefix(vector<string> &strs) {
    if (strs.empty()) return "";

    int right_most = strs[0].size() - 1;
    for (int i = 0; i < strs.size(); i++) {
      for (int j = 0; j < right_most + 1; j++) {
        if (strs[i][j] != strs[0][j]) {
          right_most = j - 1;
        }
      }
    }

    return strs[0].substr(0, right_most + 1);
  }
}

File: leetcode_38.py
class Solution:
    # @return a string
    def longestCommonPrefix(self, strs):
        if len(strs) == 0:
            return ''
        elif len(strs) == 1:
            return strs[0]

        most_right = len(strs[0])
        for i in range(1, len(strs)):
           for j in range(most_right):
               if j == len(strs[i]) or strs[i][j] != strs[i - 1][j]:
                   most_right = j
                   break

        return strs[0][0:most_right]


s = Solution()
print s.longestCommonPrefix(['12345', '1234', '123456'])
